analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereference.cs

```diff
@@ -819,5 +819,19 @@ string DoSomethingArg(IEnumerable<object> list)
             return item.ToString(); // False Negative, item could be null here
         }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/4537
+    class Repro_4537
+    {
+        private void ConditionalAccess_NullCoalescing()
+        {
+            string someString = null;
+
+            if (!someString?.Contains("a") ?? true)
+                Console.WriteLine("It's null or doesn't contain 'a'");
+            else
+                Console.WriteLine(someString.Length); // Noncompliant FP, this path is unreachable
```

Comment:
why is it unreachable? If it's not null and it does contain "a", then it will reach here

Reply:
Because `string someString = null;` and we have `NULL` constraint on that variable.

Reply:
So SE should not reach that path and should not raise S2259.
In real code base, it should raise S2583/S2589 instead.

Reply:
ah sorry I've missed the initialization at the top